From dc96ac4b6a36f9fc6da5b1432df5a3ab2ef56630 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Fri, 25 Aug 2023 01:06:05 -0700
Subject: [PATCH] It is not allowed to change IPv6 Router Address Control when
 in DHCP mode. Return error code 0xd5. 	modified:   transporthandler.cpp

---
 transporthandler.cpp | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/transporthandler.cpp b/transporthandler.cpp
index 7a99cbb..e87e184 100644
--- a/transporthandler.cpp
+++ b/transporthandler.cpp
@@ -1427,15 +1427,11 @@ RspType<> setLan(Context::ptr ctx, uint4_t channelBits, uint4_t reserved1,
                 return responseInvalidFieldRequest();
             }
 
-
-            if(control[IPv6RouterControlFlag::Static])
+            EthernetInterface::DHCPConf dhcp = channelCall<getDHCPProperty>(channel);
+            if ((dhcp == EthernetInterface::DHCPConf::v6) ||
+                (dhcp == EthernetInterface::DHCPConf::both))
             {
-                EthernetInterface::DHCPConf dhcp = channelCall<getDHCPProperty>(channel);
-                if ((dhcp == EthernetInterface::DHCPConf::v6) ||
-                    (dhcp == EthernetInterface::DHCPConf::both))
-                {
-                    return responseCommandNotAvailable();
-                }
+                return responseCommandNotAvailable();
             }
 
             IPv6RouterControlFlag::StaticControl = control[IPv6RouterControlFlag::Static];
-- 
2.25.1

